RoboticonColony/Assets/Code/HumanPlayer.cs

```diff
@@ -0,0 +1,431 @@
+﻿using System;
+
+public class HumanPlayer : AbstractPlayer
+{
+    /// <summary>
+    /// Create a HumanPlayer with the given inputController
+    /// </summary>
+    /// <param name="inputController">object to gather user input</param>
+    /// <param name="inv">a player inventory</param>
+    /// <param name="market">the market</param>
+    /// <param name="estateAgent">the estate agent</param>
+    public HumanPlayer(
+        InputController inputController,
+        Inventory inv, 
+        Market market,
+        EstateAgent estateAgent) : base(inputController, inv, market, estateAgent)
+    {
+    }
+
+    /// <summary>
+    /// The player may choose to buy a plot of land
+    /// </summary>
+    /// <param name="timeout">a time for which the phase can run</param>
+    public override void DoPhaseOne(Timeout timeout)
+    {
+        // check the player has enough money for at least the cheapest tile
+        int CheapestTilePrice = int.MaxValue;
+        foreach (Tile UnsoldTile in EstateAgent.GetAvailableTiles())
+        {
+            int CurrentTilePrice = EstateAgent.GetPrice(UnsoldTile);
+            if (CurrentTilePrice < CheapestTilePrice)
+            {
+                CheapestTilePrice = CurrentTilePrice;
+            }
+        }
+        if (Inventory.Money < CheapestTilePrice)
+        {
+            try
+            {
+                Input.Confirm("You dont have enough money to buy a tile this round", timeout);
+            }
+            catch (TimeoutException)
+            {
+                return;
+            }
+            return;
+        }
+
+        // does the player want to buy a tile?
+        bool WishesToPurchase = false;
+        try
+        {
+            WishesToPurchase = Input.PromptBool("Do you wish to purchase a tile?", timeout);
+        }
+        catch (TimeoutException)
+        {
+            return;
+        }
```

Comment:
It's getting to the point where it's hard to imagine exactly how things will run before seeing them in practice, but I'm pretty sure that the program will run both Input.Confirm, then Input.PromptBool without waiting unless we add a way for the program to wait for input before carrying on